autometa/taxonomy/ncbi.py

```diff
@@ -364,60 +345,64 @@ def parse_merged(self):
-        format='%(asctime)s : %(name)s : %(levelname)s : %(message)s',
-        datefmt='%m/%d/%Y %I:%M:%S %p',
-        level=logger.DEBUG)
-    parser = argparse.ArgumentParser(
-        description='Autometa NCBI utilities class')
-    parser.add_argument('ncbi', help='</path/to/ncbi/database/directory>')
-    parser.add_argument(
-        '--query-taxid',
-        help='query taxid to test NCBI class functionality',
-        default=1222,
-        type=int,
-    )
-    parser.add_argument('--verbose', help="add verbosity",
-                        action='store_true', default=True)
-    args = parser.parse_args()
-    ncbi = NCBI(args.ncbi, args.verbose)
-    query_taxid_parent = ncbi.parent(taxid=args.query_taxid)
-    logger.info(
-        f'{args.query_taxid} Name: {ncbi.name(taxid=args.query_taxid)}\n'
-        f'{args.query_taxid} Order: {ncbi.name(taxid=args.query_taxid, rank="order")}\n'
-        f'{args.query_taxid} Class: {ncbi.name(taxid=args.query_taxid, rank="class")}\n'
-        f'{args.query_taxid} Rank: {ncbi.rank(taxid=args.query_taxid)}\n'
-        f'{args.query_taxid} Lineage:\n{ncbi.lineage(taxid=args.query_taxid)}\n'
-        f'{args.query_taxid} is_common_ancestor {query_taxid_parent}: '
-        f'{ncbi.is_common_ancestor(parent_taxid=query_taxid_parent, child_taxid=args.query_taxid)}\n'
-    )
+        taxid_A_lineage_taxids = {taxids.get('taxid')
+                                  for taxids in self.lineage(taxid_A)}
+        taxid_B_lineage_taxids = {taxids.get('taxid')
+                                  for taxids in self.lineage(taxid_B)}
+        common_ancestor = taxid_B_lineage_taxids.intersection(
+            taxid_A_lineage_taxids)
+        common_ancestor.discard(1)  # This discards root
+        return True if common_ancestor else False
+
+    def is_valid_taxid(self, taxid):
+        """
+        Checks if the given taxid is a positive interger or not 
+
+        Parameters
+        ----------
+        `taxid` : int
+            identifer for a taxon in NCI taxonomy databses - `nodes.dmp`, `names.dmp` or `merged.dmp`
+
+        Returns
+        -------
+        `taxid` or boolean
+            `taxid` if the 'taxid' is a positive integer else False
+        """
+
+        # This check if an interger has been added as str, eg. "562"
+        if isinstance(taxid, str):
+            if taxid.isnumeric() or taxid.replace('.', '', 1).isnumeric():
+                # checks if it is something like 12.0 vs. 12.9.
+                if float(taxid).is_integer() and float(taxid).is_integer() > 0:
+                    # See https://stackoverflow.com/a/47764450/12671809
+                    return int(float(taxid))
+                else:
+                    return False
+            else:
+                return False
+        # `is_integer` returns error if only '12' is used, float(taxid) is needed
+        if not float(taxid).is_integer() or taxid < 0:
+            return False
+        return int(taxid)
```

Comment:
It appears this function has two responsibilities:

1. Determine whether the taxid can be converted to an integer.
2. _Then_ if `1` is True, convert the taxid to an integer.

We should be able to make this function more concise thereby reducing the `ReturnTypes`.

1. Check the taxid type conversion
2. Check the `taxid` (when converted) is a positive integer.
3. Return the converted taxid.

Note: If the taxid when converted is not a positive integer, you can return `0` and this would similarly evaluate to `False` in the context of other methods. This should allow you to have _only_ _one_ `return` statement.

e.g. `return taxid`